flux-refineck/src/checker.rs

```diff
@@ -583,18 +583,15 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
         &mut self,
         mut rcx: RefineCtxt,
         mut env: TypeEnv,
-        src_info: Option<SourceInfo>,
+        src_info: SourceInfo,
         target: BasicBlock,
     ) -> Result<(), CheckerError> {
         if self.is_exit_block(target) {
-            let tag = match src_info {
-                Some(info) => Tag::RetAt(info.span),
-                None => Tag::Ret,
-            };
+            let tag = Tag::RetAt(src_info.span);
             self.phase
                 .constr_gen(self.genv, &rcx, tag)
                 .check_ret(&mut rcx, &mut env, &self.output)
-                .map_err(|err| err.with_src_info_opt(src_info))
+                .map_err(|err| err.with_src_info_opt(Some(src_info)))
```

Comment:
I think we have a `with_src_info`
```suggestion
                .map_err(|err| err.with_src_info(src_info))
```